orttraining/orttraining/test/python/orttraining_test_ortmodule_api.py

```diff
@@ -154,6 +154,26 @@ def forward(self, x, y=None, z=None):
             return torch.mean(self.a) + 3 * y
         return torch.mean(self.a) + x
 
+class NeuralNetNonDifferentiableOutput(torch.nn.Module):
+    def __init__(self, input_size, hidden_size, num_classes):
+        super(NeuralNetNonDifferentiableOutput, self).__init__()
+        self.fc1 = torch.nn.Linear(input_size, hidden_size)
+        self.relu = torch.nn.ReLU()
+        self.fc2 = torch.nn.Linear(hidden_size, num_classes)
+
+    def forward(self, input1):
+        out = self.fc1(input1)
+        out1 = self.relu(out)
+        out2 = self.fc2(out1)
+        mask1 = torch.gt(out1, 0.01)
+        mask1 = mask1.long()    # TODO: Casting from bool to float or int will cause the UT failure
+                                # True is casted to 1065353216 for Cast(from=bool, to=int), whereas pytorch would give 1
```

Comment:
Interesting! Is this another bug we should raise with the exporter team?

Reply:
I haven't look into the details on the root cause of this bug... Will keep you posted :)